src/Bridge/Doctrine/DBAL/Types/TypesDumper.php

```diff
@@ -101,21 +141,28 @@ private function getTypeCode(string $classname, string $enumClass, string $type,
     if (!\class_exists($classname::class)) {
         class $classname extends \\{$baseClass}
         {
-            public const NAME = '$name';
+$code
+        }
+    }
 
-            protected function getEnumClass(): string
-            {
-                return \\{$enumClass}::class;
-            }
+PHP;
+    }
 
-            public function getName(): string
-            {
-                return static::NAME;
+    private function valueAsCode($value): string
+    {
+        if (is_string($value)) {
+            if (strpos($value, '::') !== false) { // Constant name
```

Comment:
I'm not totally happy with this heuristic since this is not bullet-proof. With this you cannot have `::` in enum values as `default`.

An alternative would be requiring a leading backslash in case of constant names but that would be less intuitive given that `types` is just fine without leading backslash. Also technically you could not have a leading backslash in enum values either anymore but that's even more of an edge case.

@ogizanagi What do you think?

Reply:
Ah indeed, that applies here too. If documented this would actually be the best solution. :+1: